### ISSUE TITLE
The relationship b/w msprime sweeps and discoal needs clarification.

### ISSUE THREAD

**Issue by molpopgen** [tskit-dev/msprime#1718]
It is currrently difficult to convert work flows from discoal to msprime because there's a scaling difference b/w the two that is not documented. (It is also technically possible that there's a time scaling error in the simulation that could give the same result, but I suspect a docs issue here.)

The short version: one has to change "s" by a factor of 2 in order to match the results.  This will lead people down a rabbit hole.

Consider the following `msprime` work flow, which writes the number of mutations to a file:

```py
import msprime
import numpy as np

Ne = 1e4
L = 1e4
ALPHA = 1000  # 4Nes
RHO = 1000  # 4Nr

hard_sweep = msprime.SweepGenicSelection(
    position=L / 2.0,
    start_frequency=1 / 2 / Ne,
    end_frequency=1 - 1 / 2 / Ne,
    s=ALPHA / 4 / Ne,
    dt=1 / 100 / Ne,
)

with open("msprime_segsites.txt", "w") as output:
    for ts in msprime.sim_ancestry(
        samples=50,
        population_size=Ne,
        model=[hard_sweep, msprime.StandardCoalescent()],
        sequence_length=L,
        recombination_rate=RHO / 4 / Ne / L,
        random_seed=666,
        discrete_genome=False,
        num_replicates=1000,
    ):
        # r = u
        ts = msprime.sim_mutations(
            ts,
            rate=RHO / 4 / Ne / L,
            model=msprime.BinaryMutationModel(),
            discrete_genome=False,
        )
        assert ts.num_samples == 100
        assert ts.num_sites == ts.num_mutations
        output.write(f"{ts.num_sites}\n")
```

Compare to the following discoal commands (master branch, cloned a few days ago):

```sh
#!/bin/bash

# Sample size is 100 b/c msprime is now defaulting to diploids, so 50 is really 100
# Alpha is 2Ns instead of the 4Ns I used in msprime to get "s"
# NOTE: I have also made these same runs w/o the -N and -i flags, and the differences are still there
../../submodules/discoal/discoal 100 100 10000  -ws 0.0 -a 500 -x 0.5 -r 1000.0 -t 1000.0   -N 10000 -i 100 -d 12345 6789 | grep segsites | cut -d" " -f 2 > discoal_segsites.txt

# Does the genic selection model differ by a factor of 2 from msprime?
../../submodules/discoal/discoal 100 100 10000  -ws 0.0 -a 250 -x 0.5 -r 1000.0 -t 1000.0   -N 10000 -i 100 -d 12345 6789 | grep segsites | cut -d" " -f 2 > discoal_segsites2.txt
```

Then, compare the results in `R` (sorry Jerome...):

```r
> x=scan("discoal_segsites.txt")
Read 100 items
> x2=scan("discoal_segsites2.txt")
Read 100 items
> y=scan("msprime_segsites.txt")
Read 1000 items
> mean(x)
[1] 4265.21
> sd(x)
[1] 169.6155
> mean(x2)
[1] 4660.56
> sd(x2)
[1] 182.6641
> mean(y)
[1] 4653.898
> sd(y)
[1] 175.0255
> ks.test(x, x2)

	Two-sample Kolmogorov-Smirnov test

data:  x and x2
D = 0.75, p-value < 2.2e-16
alternative hypothesis: two-sided

Warning message:
In ks.test(x, x2) : p-value will be approximate in the presence of ties
> ks.test(x, y)

	Two-sample Kolmogorov-Smirnov test

data:  x and y
D = 0.776, p-value < 2.2e-16
alternative hypothesis: two-sided

Warning message:
In ks.test(x, y) : p-value will be approximate in the presence of ties
> ks.test(x2, y)

	Two-sample Kolmogorov-Smirnov test

data:  x2 and y
D = 0.07, p-value = 0.7646
alternative hypothesis: two-sided

Warning message:
In ks.test(x2, y) : p-value will be approximate in the presence of ties
> 

```

**Comment by petrelharp** [comment]
ping @andrewkern

**Comment by molpopgen** [comment]
There dies indeed seem to be a scaling difference.

msprime trajectories: 

```c
static double
genic_selection_stochastic_forwards(double dt, double freq, double alpha, double u)
{
    /* this is scaled following Ewens chapter 5 e.g.,
     * w_11=1+s; w_12=1+s/2; w_22=1; that is h=0.5 */
    double ux = ((alpha / 2.0) * freq * (1 - freq)) / tanh((alpha / 2.0) * freq);
    int sign = u < 0.5 ? 1 : -1;
    return freq + (ux * dt) + sign * sqrt(freq * (1.0 - freq) * dt);
}
```

discoal:

```c
/* genicSelectionStochastic-- returns the frequency of a selected allele
which is sweeping through the population. This is the jump process
corresponding to the condition diffusion towards fixation (i.e. forwards).
takes the dt, small time increment, the current allele freq (i.e. 1 for fixation),
and alpha the selection coefficient */
double genicSelectionStochasticForwards(double dt, double currentFreq, double alpha){

	double ux = (alpha*currentFreq*(1.-currentFreq))/tanh(alpha*currentFreq);
	if (ranf() < 0.5){
	 	currentFreq += (ux * dt) + sqrt(currentFreq * (1.0 - currentFreq) * dt);
	}
	else{
		currentFreq +=  (ux * dt) - sqrt(currentFreq * (1.0 - currentFreq) * dt);
	}
	return(currentFreq);
}
```

**Comment by molpopgen** [comment]
This really should be documented.  The docs currently say that the implementation "closely follows discoal", but a key parameter has been changed.  I did read the docs that you link to, but that's not sufficient--moving a work flow from one platform to another means that "alpha" is no longer the same, which led to a lot of time figuring out why results were off.  The docs should clearly state that there's a different scaling.
